mlkem/indcpa.c

```diff
@@ -28,8 +30,16 @@
  *              polyvec *pk: pointer to the input public-key polyvec
  *              const uint8_t *seed: pointer to the input public seed
  **************************************************/
-static void pack_pk(uint8_t r[MLKEM_INDCPA_PUBLICKEYBYTES], polyvec *pk,
-                    const uint8_t seed[MLKEM_SYMBYTES]) {
+CBMC_STATIC
+void pack_pk(uint8_t r[MLKEM_INDCPA_PUBLICKEYBYTES], polyvec *pk,
+             const uint8_t seed[MLKEM_SYMBYTES])  // clang-format off
+REQUIRES(IS_FRESH(r, MLKEM_INDCPA_PUBLICKEYBYTES))
+REQUIRES(IS_FRESH(pk, sizeof(polyvec)))
+REQUIRES(IS_FRESH(seed, MLKEM_SYMBYTES))
+REQUIRES(FORALL(int, k0, 0, MLKEM_K - 1,
+  ARRAY_IN_BOUNDS(int, k1, 0, MLKEM_N - 1, pk->vec[k0].coeffs, 0, MLKEM_Q - 1)))
+ASSIGNS(OBJECT_WHOLE(r))  // clang-format on
```

Comment:
Please document those bounds in the doc string.

Reply:
fixed